sympy/solvers/solvers.py

```diff
@@ -1122,6 +1122,7 @@ def _do_dict(solution):
             solution and
             ordered_symbols and
             type(solution) is not dict and
+            type(solution) is not Piecewise and
```

Comment:
separate out Piecewise handling for solve and make new PR; don't allow list as an arg in Piecewise [recall this conversation](https://github.com/sympy/sympy/pull/12920#discussion_r126713297)